src/entity.ts

```diff
@@ -376,11 +437,45 @@ export namespace entity {
     return value instanceof entity.Key;
   }
 
+  /**
+   * Convert a protobuf `integerValue`.
+   *
+   * @private
+   * @param {object} value The `integerValue` to convert.
+   */
+  function decodeIntegerValue(value: ValueProto) {
+    const num = Number(value.integerValue);
+    if (!Number.isSafeInteger(num)) {
+      throw new Error(
+        'We attempted to return all of the numeric values, but ' +
+          (value.propertyName ? value.propertyName + ' ' : '') +
+          'value ' +
+          value.integerValue +
+          " is out of bounds of 'Number.MAX_SAFE_INTEGER'.\n" +
+          "Please consider passing 'options.wrapNumbers=true' and\n" +
+          "'options.integerTypeCastOptions' as\n" +
+          '{\n' +
+          '  integerTypeCastFunction: optionally provide <your_custom_function>\n' +
+          '  properties: optionally specify property name(s) to be cutom casted' +
+          '}\n' +
+          'to prevent this error.'
+      );
+    }
+    return num;
+  }
+
   /**
    * Convert a protobuf Value message to its native value.
    *
    * @private
    * @param {object} valueProto The protobuf Value message to convert.
+   * @param {object} typeCastOptions Config for custom `integerValue` cast.
```

Comment:
I think `typeCastOptions` is optional, so it should be bracketed. And then `typeCastOptions.integerTypeCastFunction` is required within that object, so I don't think it should be bracketed. I'm not sure that's true, though!